tests/components/onewire/test_options_flow.py

```diff
@@ -0,0 +1,237 @@
+    assert result["data_schema"].schema["device_selection"].options == {
+        "28.111111111111": False,
+        "28.222222222222": False,
+        "28.222222222223": False,
+    }
+
+    # Verify that an empty selection does not modify the options
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={INPUT_ENTRY_DEVICE_SELECTION: []},
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "device_selection"
+    assert result["errors"] == {"base": "device_not_selected"}
+
+
+async def test_user_owserver_options_set_single(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    owproxy: MagicMock,
+):
+    """Test configuring a single device."""
+    setup_owproxy_mock_devices(
+        owproxy, Platform.SENSOR, [x for x in MOCK_OWPROXY_DEVICES if "28." in x]
+    )
+
+    # Clear config options to certify functionality when starting from scratch
+    config_entry.options = {}
+
+    # Verify that first config step comes back with a selection list of all the 28-family devices
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    result = await hass.config_entries.options.async_init(config_entry.entry_id)
+    assert result["data_schema"].schema["device_selection"].options == {
+        "28.111111111111": False,
+        "28.222222222222": False,
+        "28.222222222223": False,
+    }
+
+    # Verify that a single selected device to configure comes back as a form with the device to configure
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={INPUT_ENTRY_DEVICE_SELECTION: ["28.111111111111"]},
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["description_placeholders"]["sensor_id"] == "28.111111111111"
+
+    # Verify that the setting for the device comes back as default when no input is given
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={},
+    )
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert (
+        result["data"]["device_options"]["28.111111111111"]["precision"]
+        == "temperature"
+    )
+
+
+async def test_user_owserver_options_set_multiple(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    owproxy: MagicMock,
+):
+    """Test configuring multiple consecutive devices in a row."""
+    setup_owproxy_mock_devices(
+        owproxy, Platform.SENSOR, [x for x in MOCK_OWPROXY_DEVICES if "28." in x]
+    )
+
+    # Initialize onewire hub
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Verify that first config step comes back with a selection list of all the 28-family devices
+    with patch(
+        "homeassistant.helpers.device_registry.DeviceRegistry.async_get_device",
```

Comment:
We shouldn't patch the device registry. Create a device in the device registry before starting the test instead.